crates/sui-types/src/accumulator_event.rs

```diff
@@ -22,4 +27,57 @@ impl AccumulatorEvent {
             write,
         }
     }
+
+    pub fn total_sui_in_event(&self) -> (u64 /* input */, u64 /* output */) {
+        let Self {
+            write:
+                AccumulatorWriteV1 {
+                    address: AccumulatorAddress { ty, .. },
+                    operation,
+                    value,
+                },
+            ..
+        } = self;
+
+        let TypeTag::Struct(struct_ty) = ty else {
+            return (0, 0);
+        };
+
+        if !(struct_ty.address == SUI_FRAMEWORK_ADDRESS
+            && struct_ty.module.as_str() == "balance"
+            && struct_ty.name.as_str() == "Balance")
+        {
+            // not a Balance<T>
+            return (0, 0);
+        }
+
+        debug_assert!(struct_ty.type_params.len() == 1);
+        let TypeTag::Struct(coin_type) = &struct_ty.type_params[0] else {
+            // T is not a struct type
+            return (0, 0);
+        };
+
+        if !(coin_type.address == SUI_FRAMEWORK_ADDRESS
+            && coin_type.module.as_str() == "sui"
+            && coin_type.name.as_str() == "SUI")
```

Comment:
nit constant + resolved type

Reply:
We have `GAS_STRUCT_NAME` for these--since it was added pretty much just for gas originally.